students/elmar_m/lesson05/except_exercise.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/python
+
+"""
+An exercise in playing with Exceptions.
+Make lots of try/except blocks for fun and profit.
+
+Make sure to catch specifically the error you find, rather than all errors.
+"""
+
+from except_test import fun, more_fun, last_fun
+
+
+# Figure out what the exception is, catch it and while still
+# in that catch block, try again with the second item in the list
+first_try = ['spam', 'cheese', 'mr death']
+
+try:
+    joke = fun(first_try[0])
+except NameError:
+    print('Whoops! there is no joke for:', first_try[0]) 
+    joke = fun(first_try[1])
+
+
+# Here is a try/except block. Add an else that prints not_joke
+try:
+    not_joke = fun(first_try[2])
+except SyntaxError:
+    print('Run Away!')
+else:
+    print(not_joke)
+    # next_joke = more_fun(first_try[2])
+
+# What did that do? You can think of else in this context, as well as in
+# loops as meaning: "else if nothing went wrong"
+# (no breaks in  loops, no exceptions in try blocks)
+
+# Figure out what the exception is, catch it and in that same block
+# try calling the more_fun function with the 2nd language in the list,
+# again assigning it to next_joke.
+#
+# If there are no exceptions, call the more_fun function with the last
+# language in the list regardless of whether there was an exception
+
+# Finally, while still in the try/except block and regardless of whether
+# there were any exceptions, call the function last_fun with no
+# parameters. (pun intended)
+
+langs = ['java', 'c', 'python']
+
+try:
+    more_joke = more_fun(langs[0])
+except IndexError:
+    more_joke = more_fun(langs[1])
+finally:
+    more_joke = more_fun(langs[2])
```

Comment:
You want to use an `else` block for your call to `more_fun(langs[2])` since the instructions say to call the last `more_fun` function if there are no exceptions and the `finally` clause will run whether there are any exceptions or not